dbt/adapters/databricks/connections.py

```diff
@@ -58,6 +61,27 @@
 
 logger = AdapterLogger("Databricks")
 
+
+class DbtCoreHandler(logging.Handler):
+    def __init__(self, level: Union[str, int], *args: Tuple, **kwargs: Mapping):
+        super().__init__(level=level)
+        self.logger = AdapterLogger("databricks-sql-connector")
+
+    def emit(self, record: logging.LogRecord) -> None:
+        # record.levelname will be debug, info, warning, error, or critical
+        # these map 1-to-1 with methods of the AdapterLogger
+        log_func = getattr(self.logger, record.levelname.lower())
+        log_func(record.msg)
+
+
+pysql_logger_level = os.environ.get("DBT_DATABRICKS_CONNECTOR_LOG_LEVEL", "INFO").upper()
+pysql_handler = DbtCoreHandler(level=pysql_logger_level)
+
+pysql_logger = logging.getLogger("databricks.sql")
+pysql_logger.setLevel(logging.DEBUG)
```

Comment:
I'm confused why you default the pysql_handler to INFO (based on env variable), but then set the psyql_logger to DEBUG.

Reply:
Does this only turn on the pysql_logger if the level is DEBUG, or does this overwrite what was done on line 78?

Reply:
No it doesn't overwrite anything. The logger and the handler each have a level setting. Sometimes a handler should send an email but only for messages at `CRITICAL` level. Whereas other handlers need to capture every possible log message (`DEBUG` level).

Changing the level of the logger itself means the events never even reach the handler if they don't meet or exceed its `level`.

The reason the code was different initially is that I figured we should just turn the pysql logger onto the loudest setting (`DEBUG`) and then use the environment variable to control which messages the DbtCoreHandler actually passes to its AdapterLogger.

It's not a major difference. So I changed both the logger and the handler to use the same level.